validator/sawtooth_validator/state/state_view.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+from sawtooth_validator.state.merkle import MerkleDatabase
+
+
+class StateViewFactory(object):
+    """The StateViewFactory produces StateViews for a particular merkle root.
+
+    This factory produces read-only views of a merkle tree. For a given
+    database, these views are considered immutable.
+    """
+
+    def __init__(self, database):
+        """Initializes the factory with a given database.
+
+        Args:
+            database (:obj:`Database`): the database containing the merkle
+                tree.
+        """
+        self._database = database
+
+    def create_view(self, state_root_hash):
+        """Creates a StateView for the given state root hash.
+
+        Returns:
+            StateView: state view locked to the given root hash.
+        """
+        return StateView(MerkleDatabase(self._database, state_root_hash))
+
+
+class StateView(object):
+    """The StateView is a snapshot of state stored in a merkle tree.
```

Comment:
Perhaps: "The StateView provides read only access to the state stored at a particular merkle root."